src/main/kotlin/first_pass/node_parser/MemberDeclarationParser.kt

```diff
@@ -0,0 +1,46 @@
+package first_pass.node_parser
+
+import cadet_model.CadetClass
+import cadet_model.CadetMember
+import cadet_model.CadetMemberType
+import cadet_model.CadetParameter
+import com.github.javaparser.ast.Node
+import com.github.javaparser.ast.body.ConstructorDeclaration
+import com.github.javaparser.ast.body.MethodDeclaration
+import com.github.javaparser.ast.body.Parameter
+
+object MemberDeclarationParser : AbstractNodeParser() {
+
+    fun instantiateMethod(node: MethodDeclaration, parent: CadetClass): CadetMember {
+        return CadetMember().apply {
```

Comment:
I did this on purpose because it avoided messy casting and if-statements. 
I extracted the shared parts into a separate private 'instantiateBaseMember' function, and left parts which are Node-specific to be handled by their respective functions.